lib/rubocop/cop/rspec/rails/minitest_assertions.rb

```diff
@@ -229,6 +252,45 @@ def assertion
               'be_empty'
             end
           end
+
+          MSG = 'Use `%<prefer>s`.'
+
+          ASSERTION_MATCHERS = constants(false).filter_map do |c|
+            const = const_get(c)
+
+            const if const.is_a?(Class) && const.superclass == BasicAssertion
+          end
+
+          RESTRICT_ON_SEND = ASSERTION_MATCHERS.flat_map { |m| m::MATCHERS }
+
+          ASSERTION_MATCHERS.each do |m|
+            name = m.name.split('::').last
+
+            def_node_matcher "minitest_#{name}".to_sym, m::NODE_MATCHER_PATTERN
```

Comment:
Wondering if it’s possible to define node matchers in assertion classes

Reply:
It would mean that they'd have to extend from Rubocop itself in some way but that might not be the end of the world - however it would also mean a greater refactor because the node matcher gets defined on the instance rather than as a static method, so the whole constructor would need to change.

If it's ok with you, I'd prefer to stick with the current refactor since that is working and (in my opinion) is pretty optimal - I'm happy to explore a further refactor otherwise though.